spec/helpers/application_helper/toolbar_builder_spec.rb

```diff
@@ -2053,6 +2061,126 @@ def setup_firefox_with_linux
+            end
+          end
+
+          context "depot is configured" do
+            it "doesn't disable button" do
+              @record.status = "started"
+              @record.log_file_depot = file_depot
+              expect(build_toolbar_disable_button("collect_logs")).to eq(false)
+            end
+          end
+        end
+      end
+    end
+
+    context "when record class = Zone" do
+      let(:log_file) { FactoryGirl.create(:log_file) }
+      let(:miq_task) { FactoryGirl.create(:miq_task) }
+      let(:file_depot) { FactoryGirl.create(:file_depot) }
+      let(:miq_server) { FactoryGirl.create(:miq_server) }
+
+      before do
+        @record = FactoryGirl.create(:zone)
+      end
+
+      context "id=zone_collect_logs" do
+        context "no server of zone is started" do
+          it "disables button with correct title" do
+            miq_server.status = "not responding"
+            @record.miq_servers << miq_server
+            error_msg = "Cannot collect current logs unless there are started Servers in the Zone"
+            expect(build_toolbar_disable_button("zone_collect_logs")).to eq(error_msg)
+          end
+        end
+
+        context "log collecting is in progress, log_file.state = collection" do
+          it "disables button with correct title" do
+            log_file.resource = @record
+            log_file.state = "collecting"
+            log_file.save
+            miq_server.log_files << log_file
+            miq_server.status = "started"
+            @record.miq_servers << miq_server
+            @record.log_file_depot = file_depot
+            error_msg = "Log collection is already in progress for this Server"
+            expect(build_toolbar_disable_button("zone_collect_logs")).to eq(error_msg)
+          end
+        end
+
+        context "log collecting is in progress, MiqTask.status != Finished" do
+          it "disables button with correct title" do
+            miq_server.status = "started"
+            @record.miq_servers << miq_server
+            @record.log_file_depot = file_depot
+            miq_task.name = "Zipped log retrieval for XXX"
+            miq_task.miq_server_id = miq_server.id
+            miq_task.save
+            error_msg = "Log collection is already in progress for one or more Servers in this Zone"
+            expect(build_toolbar_disable_button("zone_collect_logs")).to eq(error_msg)
+          end
+        end
+
+        context "depot is not configured" do
+          it "disables button with correct title" do
+            miq_server.status = "started"
+            @record.miq_servers << miq_server
+            @record.log_file_depot = nil
+            error_msg = "This Zone do not have Log Depot settings configured, collection not allowed"
+            expect(build_toolbar_disable_button("zone_collect_logs")).to eq(error_msg)
+          end
+        end
+
+        context "depot is configured" do
+          it "doesn't disable button" do
```

Comment:
same as above, too many uneeded indents, let's remove the extra contexts with a single it.